src/main/scala/com/github/traviscrawford/spark/dynamodb/BaseScanner.scala

```diff
@@ -52,17 +54,31 @@ private[dynamodb] trait BaseScanner {
       .withTotalSegments(config.totalSegments)
       .withSegment(config.segment)
 
-    config.maybeRequiredColumns
-      .foreach(requiredColumns => scanSpec.withProjectionExpression(requiredColumns.mkString(",")))
+    // Parse any projection expressions passed in
+    val exprSpecBuilder = config.maybeRequiredColumns
+      .map(requiredColumns => new ExpressionSpecBuilder().addProjections(requiredColumns: _*))
+    val parsedProjectionExpr = exprSpecBuilder.map(xSpecBuilder => xSpecBuilder.buildForScan())
+    val projectionNameMap = parsedProjectionExpr.flatMap(projExpr => Option(projExpr.getNameMap))
+      .map(_.asScala.toMap).getOrElse(Map.empty)
+    val projectionValueMap = parsedProjectionExpr.flatMap(projExpr => Option(projExpr.getValueMap))
+      .map(_.asScala.toMap).getOrElse(Map.empty)
+    parsedProjectionExpr.foreach(expr =>
+      scanSpec.withProjectionExpression(expr.getProjectionExpression))
+
     // Parse any filter expression passed in as an option
-    config.maybeFilterExpression.map(filterExpression => ParsedFilterExpression(filterExpression))
-      .foreach(parsedExpr => {
-        scanSpec.withFilterExpression(parsedExpr.expression)
-        Option(parsedExpr.expressionNames).filter(_.nonEmpty)
-          .foreach(exprNames => scanSpec.withNameMap(exprNames.asJava))
-        Option(parsedExpr.expressionValues).filter(_.nonEmpty)
-          .foreach(exprValues => scanSpec.withValueMap(exprValues.asJava))
-      })
+    val parsedFilterExpr = config.maybeFilterExpression
+      .map(filterExpression => ParsedFilterExpression(filterExpression))
+    val filterNameMap = parsedFilterExpr.map(_.expressionNames).getOrElse(Map.empty)
+    val filterValueMap = parsedFilterExpr.map(_.expressionValues).getOrElse(Map.empty)
+    parsedFilterExpr.foreach(expr =>
+      scanSpec.withFilterExpression(expr.expression))
+
+    // Combine parsed name and value maps from the projections and filter expressions
+    val nameMap = projectionNameMap ++ filterNameMap
+    Option(nameMap).filter(_.nonEmpty).foreach(nMap => scanSpec.withNameMap(nMap.asJava))
+    val valueMap = projectionValueMap ++ filterValueMap
+    Option(valueMap).filter(_.nonEmpty).foreach(vMap => scanSpec.withValueMap(vMap.asJava))
```

Comment:
`ScanSpec` uses unmodifiable maps under the covers, so I couldn't just add the name and value maps as I went along. Instead, I combine them here for both the projection and filter expressions and add the result to the spec.